### PR TITLE
Logging Code Snippet

### PR DESCRIPTION
**Pull request by danewalton**
Added a code snippet to the README detailing how one might use logging. Basically copied over from a doc I was going to pass to the IoT team to show how we would use it.

### CODE REVIEW

File: sdk/core/core/README.md [hunk 1]
```diff
@@ -64,7 +64,71 @@ az_span hello_world = AZ_SPAN_FROM_STR("Hello world!");
...
+  az_log_write(AZ_LOG_HTTP_REQUEST, test_log_message);
+}
+
+
+/* User Application Code */
+az_log_classification const classifications[] = { AZ_LOG_HTTP_REQUEST, AZ_LOG_HTTP_RESPONSE };
+
+void test_log_func(az_log_classification classification, az_span message)
+{
+    printf("%.*s", az_span_length(message), az_span_ptr(message));
```
Comment (antkmsft) [inline sdk/core/core/README.md]: I recommend to add newline at the end of message, `"%.*s\n"`.

Comment (danewalton) [reply sdk/core/core/README.md]: addressed

Comment (antkmsft) [reply sdk/core/core/README.md]: Or maybe even `fprintf(stderr, "%.*s\n", (unsigned int)az_span_length(message), az_span_ptr(message));`, if that's not an unneccessary complication in your opinion.

Comment (danewalton) [reply sdk/core/core/README.md]: I think unless others are passionate about it I'll probably leave it. Just to have the most fundamentally basic example.

Comment (danewalton) [reply sdk/core/core/README.md]: open to other opinions though. I'm definitely not married to my idea

Comment (antkmsft) [reply sdk/core/core/README.md]: I wasn't sure, and now I think I agree with you. Let's keep it simple.

Comment (antkmsft) [reply sdk/core/core/README.md]: BTW, there's currently seem to be some intermittent issues with the CI/CD pipeline, you probably don't see it. I'll rerun the checks and merge your PR.

**Pull request merged**